core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/CoreDocumentSynchronizationConfig.java

```diff
@@ -156,17 +158,44 @@ public void setStale(final boolean stale) {
     }
   }
 
+  /**
+   * In the event that an irrecoverable error has occurred,
+   * updates should no longer be processed for a document.
+   * The config should reflect that it is in a frozen state.
+   *
+   * When a new write operation happens to this document locally
+   * we should unfreeze the document and resume trying to sync it.
+   *
+   * @param isFrozen whether or not this config is frozen
+   */
+  void setFrozen(final boolean isFrozen) {
```

Comment:
[major] In general setting anything must be reflected by taking a write lock out on the document and updating the collection with this document with frozen now being set. See setStale